src/circuit/auto-relay.js

```diff
@@ -231,8 +231,7 @@ class AutoRelay {
 
     // Try to listen on known peers that are not connected
     for (const peerId of knownHopsToDial) {
-      const connection = await this._libp2p.dial(peerId)
```

Comment:
Can you please add a test to exercise this code path ? Or if it's a problem with codecov, maybe add a codecov specific comment.

Reply:
Yes, this is codecov having a bad time.. This [test](https://github.com/libp2p/js-libp2p/blob/v0.31.0-rc.3/test/relay/auto-relay.node.js#L333) will exercise this code path

Reply:
I will add a sinon check in that test